addons/mrp/models/mrp_bom.py

```diff
@@ -476,6 +476,36 @@ def _update_order_line_info(self, product_id, quantity, child_field=False, **kwa
 
         return self.env['product.product'].browse(product_id).standard_price
 
+    # -------------------------------------------------------------------------
+    # DOCUMENT
+    # -------------------------------------------------------------------------
+
+    def _get_mail_thread_data_attachments(self):
+        res = super()._get_mail_thread_data_attachments()
+        return res | self._get_extra_attachments()
+
+    def _get_extra_attachments(self):
+        final_domain = []
+        bom_domain = ['&', ('attached_on_mrp', '=', 'bom'), ('active', '=', 't')]
+        is_byproduct = self.env.user.has_group('mrp.group_mrp_byproducts')
+        for bom in self:
+            product_subdomain = ['|',
+                '&', ('res_model', '=', 'product.product'), ('res_id', '=', bom.product_id.id),
+                '&', ('res_model', '=', 'product.template'), ('res_id', '=', bom.product_tmpl_id.id)]
+            if is_byproduct:
+                product_domain = OR([product_subdomain, [
+                    '|',
+                    '&', ('res_model', '=', 'product.product'), ('res_id', '=', bom.byproduct_ids.product_id.id),
+                    '&', ('res_model', '=', 'product.template'), ('res_id', '=', bom.byproduct_ids.product_id.product_tmpl_id.id)]])
+            else:
+                product_domain = product_subdomain
+            prod_final_domain = AND([bom_domain, product_domain])
+            final_domain = OR([final_domain, prod_final_domain]) if final_domain else prod_final_domain
+
```

Comment:
what about document on components (`move_raw_ids.product_id`) ?

Reply:
The ``_get_extra_attachments`` method is only used to display documents of the main template/product of the bom in the chatter, so I think this is right